public/app/features/plugins/components/AppRootPage.tsx

```diff
@@ -81,16 +82,56 @@ export function AppRootPage({ pluginId, pluginNavSection }: Props) {
     />
   );
 
+  // Because of the fallback at plugin routes, we need to check
+  // if the user has permissions to see the plugin page.
+  const userHasPermissionsToPluginPage = (path: string) => {
+    if (!plugin.meta?.includes) {
+      return true;
+    }
+    let rolesMap = new Map();
```

Comment:
It seems like this logic can work without this map here. Perhaps you can use this same loop to check for what you need?